### ISSUE TITLE
Error while creating custom action. Suggestion: remove entity_id requirement when using YAML-mode

### ISSUE THREAD

**Issue by thrawnarn** [nielsfaber/alarmo#185]
I want to be able to send mqtt messages from an action. But as it is now it is possible as I need to define an entity_id under service_data and the mqtt service won't accept entity_id in the data. My suggestion is to remove the validation for entity_id when using the YAML-mode. 

Problem in pictures
When using entity_id:
![image](https://user-images.githubusercontent.com/5276026/115672629-0912c880-a34c-11eb-850e-98530cd60a26.png)
and pressing try it, the mqtt service returns this:
![image](https://user-images.githubusercontent.com/5276026/115672732-22b41000-a34c-11eb-855f-65fa0ddc5e91.png)

When removing entity_id:
![image](https://user-images.githubusercontent.com/5276026/115672794-32cbef80-a34c-11eb-95ce-674cf17d3023.png)
and pressing try it, the alarmo validation returns this:
![image](https://user-images.githubusercontent.com/5276026/115672853-411a0b80-a34c-11eb-99ad-95e94c3517d9.png)

**Comment by nielsfaber** [comment]
Alarmo already has a MQTT interface, see here: https://github.com/nielsfaber/alarmo#mqtt
The actions editor is intended for switching devices (like a siren), not for what you're trying.
If you want to create more advanced automations, you may want to use standard HA automations instead of the Alarmo actions panel.

**Comment by thrawnarn** [comment]
First of, I totally understand if this is not in scope for alarmo, but I get a feeling you misunderstood what I'm trying todo here. 
First of I'm trying to publish to specific mqtt topics, as I understand that is not possible with current MQTT implementation right?

I really think what I'm trying todo is in the scope of something simple, but I could of course be wrong :)

I want to use a simple [alarm keypad](https://www.zigbee2mqtt.io/devices/URC4450BC0-X-R.html#xfinity-urc4450bc0-x-r) that supports different states. So when the alarm is pending I want to let the keypad know that it is pending, when it is armed I want to let the keypad know that. It's almost as simple as turning a siren on or off, the only difference is that I need to do it with mqtt. 

As you point out I could do this with normal automations, but it would be great to be able to do this within alarmo and I don't think I'm the only one who would want this.

Now that I explained the purpose more I will of course accept if you still don't think it's inside of Alarmos scope.

**Comment by nielsfaber** [comment]
That's a cute keypad, it looks like a good companion for Alarmo.
Although the MQTT interface of Alarmo is _intended_ to be compatible for such use-cases, it has proven to be really difficult to be generic.
In your case you need a quite specific JSON formatted payload to be sent by Alarmo, which is currently not possible.
Unless you have node-red installed, in that case you could easily create some task that converts Alarmo data to the format your device requires.

I'm a zigbee2mqtt user myself, but never had to interact directly through MQTT.
Would have been easier if the integration came with separate entities (like switches) for the arming and disarming, instead of having to do this low-level communication.

Anyway, I agree that this job is simple (I indeed thought you wanted to do something exotic with it) and Alarmo should provide functionality for it.
However the built-in Actions panel is the most poorly maintained part of Alarmo, since I prioritise the alarm functions.
I see many improvement points (many of which are open feature requests here), more than I can keep up with actually.
If it would help you if I make the entity_ID field optional, it can be done (in fact, this already the case for the notification editor). 

For the time being creating a HA automation is your best (and only) option.

**Comment by thrawnarn** [comment]
well, it wasn't that difficult using HA automations (I don't want to use node-red, I want as much as possible contained inside HA). For anyone else who want to integrate this and got the same problem I will just paste my three automations handeling the keypad here...

```
- id: '1618259530095'
  alias: Handle Alarm Keypad Command
  description: ''
  trigger:
  - platform: mqtt
    topic: zigbee2mqtt/hallway_alarm_keypad
  action:
  - variables:
      action: '{{ trigger.payload_json.action }}'
      code: '{{ trigger.payload_json.action_code }}'
  - choose:
    - conditions:
      - condition: template
        value_template: '{{ action ==  ''disarm''}}'
      sequence:
      - service: alarm_control_panel.alarm_disarm
        data:
          entity_id: alarm_control_panel.alarmo
          code: '{{ code }}'
    - conditions:
      - condition: template
        value_template: '{{ action == ''arm_all_zones'' }}'
      sequence:
      - service: alarm_control_panel.alarm_arm_away
        data:
          entity_id: alarm_control_panel.alarmo
          code: '{{ code }}'
    - conditions:
      - condition: template
        value_template: '{{ action == ''arm_night_zones'' }}'
      sequence:
      - service: alarm_control_panel.alarm_arm_night
        data:
          entity_id: alarm_control_panel.alarmo
          code: '{{ code }}'
    - conditions:
      - condition: template
        value_template: '{{ action == ''arm_day_zones'' }}'
      sequence:
      - service: alarm_control_panel.alarm_arm_home
        data:
          entity_id: alarm_control_panel.alarmo
          code: '{{ code }}'
  mode: single
- id: '1619097407751'
  alias: Handle Alarm Keypad Status
  description: ''
  trigger:
  - platform: state
    entity_id: alarm_control_panel.alarmo
  condition: []
  action:
  - variables:
      new_keypad_mode: '{% if( trigger.to_state.state == ''disarmed'' ) %} disarm
        {% elif( trigger.to_state.state == ''arming'' and trigger.to_state.attributes.arm_mode
        == ''armed_away'' ) %} arming_away {% elif( trigger.to_state.state == ''arming''
        and trigger.to_state.attributes.arm_mode == ''armed_home'' ) %} arming_stay
        {% elif( trigger.to_state.state == ''arming'' and trigger.to_state.attributes.arm_mode
        == ''armed_night'' ) %} arming_night {% elif( trigger.to_state.state == ''arming''
        ) %} exit_delay {% elif( trigger.to_state.state == ''triggered'' ) %} in_alarm
        {% elif( trigger.to_state.state == ''pending'' ) %} entry_delay {% elif( trigger.to_state.state
        == ''armed_away'' ) %} arm_all_zones {% elif( trigger.to_state.state == ''armed_night'')
        %} arm_night_zones {% elif( trigger.to_state.state == ''armed_home'') %} arm_day_zones
        {% elif( trigger.to_state.state == ''armed_custom_bypass'') %} arm_all_zones
        {% else %} not_ready {% endif %}'
  - service: mqtt.publish
    data:
      topic: zigbee2mqtt/hallway_alarm_keypad/set
      payload_template: '{ "arm_mode": { "mode": "{{new_keypad_mode}}" } }'
  mode: restart
- id: '1619100284340'
  alias: Handle Alarm Keypad Error
  description: ''
  trigger:
  - platform: event
    event_type: alarmo_event
  condition:
  - condition: template
    value_template: '{{ trigger.event.data.event == ''failed_to_arm'' }}'
  action:
  - service: mqtt.publish
    data:
      topic: zigbee2mqtt/hallway_alarm_keypad/set
      payload_template: '{ "arm_mode": { "mode": "not_ready" } }'
  - delay:
      hours: 0
      minutes: 0
      seconds: 20
      milliseconds: 0
  - service: mqtt.publish
    data:
      topic: zigbee2mqtt/hallway_alarm_keypad/set
      payload_template: '{ "arm_mode": { "mode": "disarm" } }'
  mode: single
```

**Comment by nielsfaber** [comment]
This should be solved in v1.6.0.
YAML configuration is more flexible now.
